runner/internal/runner/api/http.go

```diff
@@ -82,8 +82,8 @@ func (s *Server) runPostHandler(w http.ResponseWriter, r *http.Request) (interfa
 	var runCtx context.Context
 	runCtx, s.cancelRun = context.WithCancel(context.Background())
 	go func() {
-		_ = s.executor.Run(runCtx) // todo handle error
-		s.jobBarrierCh <- nil      // notify server that job finished
+		s.executor.Run(runCtx)
```

Comment:
Note the linter error.
Btw, I'd suggest we make the runner-test workflow fail is there are any errors